_layouts/post.html

```diff
@@ -49,10 +49,46 @@ <h1 class="post-title" itemprop="name headline">{{ page.title }}</h1>
 </script>
 <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.12.2/jquery.min.js"></script>
 <script src="https://rawgit.com/RickStrahl/jquery-resizable/master/dist/jquery-resizable.min.js"></script>
-<script src="https://cdn.rawgit.com/play-with-docker/sdk/87c86f26/dist/pwd.js"></script>
+<script src="https://cdn.rawgit.com/play-with-docker/sdk/master/dist/pwd.js"></script>
 <script src="{{site.baseurl}}/js/quiz.js"></script>
 <script>
+        !function(){var analytics=window.analytics=window.analytics||[];if(!analytics.initialize)if(analytics.invoked)window.console&&console.error&&console.error("Segment snippet included twice.");else{analytics.invoked=!0;analytics.methods=["trackSubmit","trackClick","trackLink","trackForm","pageview","identify","reset","group","track","ready","alias","debug","page","once","off","on"];analytics.factory=function(t){return function(){var e=Array.prototype.slice.call(arguments);e.unshift(t);analytics.push(e);return analytics}};for(var t=0;t<analytics.methods.length;t++){var e=analytics.methods[t];analytics[e]=analytics.factory(e)}analytics.load=function(t){var e=document.createElement("script");e.type="text/javascript";e.async=!0;e.src=("https:"===document.location.protocol?"https://":"http://")+"cdn.segment.com/analytics.js/v1/"+t+"/analytics.min.js";var n=document.getElementsByTagName("script")[0];n.parentNode.insertBefore(e,n)};analytics.SNIPPET_VERSION="4.0.0";
+        analytics.load("GvscnIMX05fG8s2MEZY7PkSV4BTsFvOU");
+        analytics.page();
+        }}();
+</script>
+<script>
+    function getUser() {
+      var terms = {{ terms }}
+      if(terms>0) {
+        var target = document.getElementsByClassName('xterm-rows');
+        if (target.length === 0) {
+            setTimeout(getUser,1000);
```

Comment:
This logic is to identify when the user is not logged in and we are basically waiting until he is to identify him?

Reply:
basically yes. If there's a terminal, it'll wait until the terminal is populated, which only happens after a login. That's the only way I can figure out if they're logged in. If there's no terminal,  it goes ahead and checks if they are logged in anyway.

Reply:
I think we should provide in the sdk a nicer way to do this. Would something like: 
```
pwd.on('init', function() {
  // callback is called as soon as a pwd session has been initiated
});
```
This way you don't need to wait for specific html dom elements to exist.
What do you think @marcosnils? Should be fairly easy to add something like that.